dags/synapse-datasets-created-7-days.py

```diff
@@ -0,0 +1,221 @@
+"""This script executes a query on Snowflake to retrieve data about entities created in the past 7 days and stores the results in a Synapse table.
+It is scheduled to run every Monday at 00:00 UTC.
+
+DAG Parameters:
+- `snowflake_conn_id`: The connection ID for the Snowflake connection.
+- `synapse_conn_id`: The connection ID for the Synapse connection.
+- 'current_date_time': The current date time in UTC timezone
+- `backfill`: Whether to backfill the data. Defaults to `False`.
+- `backfill_date_time`: The date time to backfill the data from. Will be ignored if `backfill` is `False`.
+"""
+
+from dataclasses import dataclass
+from datetime import date, datetime, timezone
+from typing import List
+
+import synapseclient
+from airflow.decorators import dag, task
+from airflow.models import Variable
+from airflow.models.param import Param
+from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
+from orca.services.synapse import SynapseHook
+from slack_sdk import WebClient
+
+dag_params = {
+    "snowflake_conn_id": Param("SNOWFLAKE_SYSADMIN_PORTAL_RAW_CONN", type="string"),
+    "synapse_conn_id": Param("SYNAPSE_ORCA_SERVICE_ACCOUNT_CONN", type="string"),
+    "current_date_time": Param(
+        datetime.now(timezone.utc).strftime("%Y-%m-%d  %H:%M:%S"), type="string"
+    ),
+    "backfill": Param(False, type="boolean"),
+    # backfill_date_time string format: YYYY-MM-DD HH:MM:SS
+    "backfill_date_time": Param("1900-01-01 00:00:00", type="string"),
+}
+
+dag_config = {
+    "schedule_interval": "0 0 * * 1",
+    "start_date": datetime(2025, 2, 1),
+    "catchup": False,
+    "default_args": {
+        "retries": 1,
+    },
+    "tags": ["snowflake"],
+    "params": dag_params,
+}
+
+SYNAPSE_RESULTS_TABLE = "syn64919238"
+
+
+@dataclass
+class EntityCreated:
+    """Dataclass to record
+
+    Attributes:
+        name: The name of this entity.
+        id: The unique immutable ID for this entity.
+        parent_id: parent id of the entity
+        node_type: Type of Node
+        content_type: Content type of folder annotated,
+        created_on: The date this entity was created.
+        created_by: The ID of the user that created this entity.
+        is_public: True to indicate if this entity is public.
+    """
+
+    name: str
+    id: int
+    parent_id: int
+    node_type: str
+    content_type: str
+    created_on: str
+    user_name: str
```

Comment:
`user_name` isn't included in the docstring of the class.

Also, this is a little mis-leading. It isn't actually the username, it's their `full_name`, or `first_last_name`. This, along with the query should probably be updated (And maybe the Synapse table column name?)

Relevant line of SQL:
`CONCAT(first_name,' ',last_name) AS user_name,`

Reply:
Thanks Bryan! `full_name` is reasonable.